parquet-hadoop/src/main/java/parquet/filter2/statisticslevel/StatisticsFilter.java

```diff
@@ -67,13 +67,14 @@ private ColumnChunkMetaData getColumnChunk(ColumnPath columnPath) {
   }
 
   // is this column chunk composed entirely of nulls?
+  // in case of all nulls, the stats object read from file metadata is empty
   private boolean isAllNulls(ColumnChunkMetaData column) {
-    return column.getStatistics().getNumNulls() == column.getValueCount();
+    return (column.getStatistics().isEmpty()) || (column.getStatistics().getNumNulls() == column.getValueCount());
```

Comment:
I think we should fix it in the write path and have column.getStatistics().getNumNulls() == column.getValueCount() in that case.
missing statistics should mean "the writer did not have that feature"
We should not have empty row groups. There was a bug in the past that produced empty row groups in some cases but this was fixed.